File: src/quest_editor/actions/EditIdAction.ts
import { QuestEditAction } from "./QuestEditAction";

export class EditIdAction extends QuestEditAction<number> {
    readonly description = "Edit ID";

    undo(): void {
        this.quest.set_id(this.old);
    }

    redo(): void {
        this.quest.set_id(this.new);
    }
}

File: web/src/main/kotlin/world/phantasmal/web/questEditor/actions/EditIdAction.kt
package world.phantasmal.web.questEditor.actions

import world.phantasmal.web.core.actions.Action
import world.phantasmal.web.questEditor.models.QuestModel

class EditIdAction(
    private val quest: QuestModel,
    private val newId: Int,
    private val oldId: Int,
) : Action {
    override val description: String = "Edit ID"

    override fun execute() {
        quest.setId(newId)
    }

    override fun undo() {
        quest.setId(oldId)
    }
}
